internal/versions/api.go

```diff
@@ -73,25 +75,145 @@ func GetModelVersions(v Versions) models.Versions {
+	if !swag.BoolValue(onlyLatest) {
+		return releaseImages, nil
+	}
+
+	// Get all relevant major.minor openshift versions
+	releaseImagesMajorMinorVersionsSet := map[string]bool{}
+	for _, releaseImage := range releaseImages {
+		// Typically, releaseImage.OpenshiftVersion follow the major.minor format, though exceptions exist with static release images.
+		majorMinorVersion, err := common.GetMajorMinorVersion(*releaseImage.OpenshiftVersion)
+		if err != nil {
+			return nil, err
+		}
+
+		releaseImagesMajorMinorVersionsSet[*majorMinorVersion] = true
+	}
+
+	filteredReleaseImages := models.ReleaseImages{}
+	for majorMinorVersion := range releaseImagesMajorMinorVersionsSet {
+		latestReleaseImageForMajorMinor, err := getLatestReleaseImageForMajorMinor(releaseImages, majorMinorVersion)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error occurred while trying to get the latest release image for version: %s", majorMinorVersion)
+		}
+		filteredReleaseImages = append(filteredReleaseImages, latestReleaseImageForMajorMinor)
+	}
+
+	return filteredReleaseImages, nil
+}
+
+func filterReleaseImagesByVersion(releaseImages models.ReleaseImages, versionPattern *string) models.ReleaseImages {
+	if versionPattern == nil {
+		return releaseImages
+	}
+
+	filterdReleaseImagesInterface := funk.Filter(releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		return strings.Contains(*releaseImage.Version, *versionPattern)
+	})
+
+	return filterdReleaseImagesInterface.([]*models.ReleaseImage) // models.ReleaseImages does not work for convertion
+}
+
+func filterIgnoredVersions(log logrus.FieldLogger, releaseImages models.ReleaseImages, ignoredVersions []string) models.ReleaseImages {
+	return funk.Filter(releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		version := strings.TrimSuffix(*releaseImage.Version, "-multi")
+
+		majorMinorVersion, err := common.GetMajorMinorVersion(version)
+		if err != nil {
+			log.WithError(err).Debugf("error occurred while trying to get the major.minor version of '%s', filtering this version out", version)
+			return false
+		}
+
+		return !funk.Contains(ignoredVersions, version) && !funk.Contains(ignoredVersions, *majorMinorVersion)
+	}).([]*models.ReleaseImage)
+}
+
+func getReleaseImageSupportLevel(releaseImage *models.ReleaseImage) (*string, error) {
+	if releaseImage.SupportLevel != "" {
+		return &releaseImage.SupportLevel, nil
+	}
+
+	isPreRelease, err := common.IsVersionPreRelease(*releaseImage.Version)
+	if err != nil {
+		return nil, err
+	}
+
+	if *isPreRelease {
+		return swag.String(models.ReleaseImageSupportLevelBeta), nil
+	}
+
+	return swag.String(models.ReleaseImageSupportLevelProduction), nil
+}
+
 func (h *apiHandler) V2ListSupportedOpenshiftVersions(ctx context.Context, params operations.V2ListSupportedOpenshiftVersionsParams) middleware.Responder {
+	handler := h.versionsHandler.(*restAPIVersionsHandler)
```

Comment:
Instead of asserting the type here why not use the actual type in `apiHandler`?

Reply:
In second thought, I think it makes more sense to use Handler and assert, as it makes much more sense to test the api handler this way, because I can use a mock.

Reply:
Okay, but just in case we end up ever using these APIs in kube mode maybe check that the type assertion was successful here.